### ISSUE TITLE
Default example for `Tooltip` is inaccessible

### ISSUE THREAD

**Issue by khiga8** [primer/react#2595]
The default example for the `Tooltip` doc is on a static element, which we really want to discourage. Tooltip on a static element is one of the most common accessibility issues we're seeing in dotcom, and it's usually a design problem that can't easily be fixed.

Until https://github.com/github/primer/issues/1244 is implemented, I propose we update this example to use an interactive element, emphasize that it should not be set on static elements, and link to [Tooltip alternatives.](https://primer.style/design/accessibility/tooltip-alternatives/).

**Comment by lesliecdubs** [comment]
Thanks @khiga8! Dropping a link to the Tooltip docs page for reference: https://primer.style/react/Tooltip

**Comment by mperrotti** [comment]
@khiga8 - what do you mean by "static element"? Do you mean that it's an element that isn't focusable?

**Comment by khiga8** [comment]
@mperrotti Yep! By static, I meant non-interactive, non-focusable.

**Comment by khiga8** [comment]
I keep saying static but non-interactive is probably a better term :D

**Comment by khiga8** [comment]
From [Tooltip alternatives](https://primer.style/design/accessibility/tooltip-alternatives):

> Tooltips should never be set on non-interactive elements (for example, `div`, `span`, `p`), and should only ever be set on interactive elements (for example, `button`, `a`). Tooltips on non-interactive elements are not accessible to keyboard users and screen reader users.

**Comment by mperrotti** [comment]
@khiga8 - is it ok to put a tooltip on a non-interactive element that has `tabindex=0`? That would at least make it focusable

**Comment by khiga8** [comment]
@mperrotti No, tooltip must be on a natively interactive control. I had the same question when investigating tooltip a while back, and here's a bit more context from @jscholes on why it would not be appropriate [from an internal issue comment](https://github.com/github/accessibility/issues/331#issuecomment-952157307): 

> tooltip triggering should be limited to proper interactive, focusable controls.  If they are triggered on a `span`, `div`, `p` or whatever, making those focusable doesn't really fix any of the problems, apart from making them reachable from the keyboard.  But for users who arent sequentially navigating with Tab, the behaviour is still completely non-obvious, e.g.:


> 1. most screen reader users will never be able to trigger the announcement of the accessible description, because there is no interactive element for them to navigate directly to;
> 2. if the text is in the DOM after the textual element, there is nothing to semantically link what will just be two pieces of plain text;
> 3. if the `span` or whatever doesn't look interactive, a speech recognition user isn't likely to ask their software to focus it;
> 4. ... etc.

.. and a follow up:

> > Just to clarify one thing that I wrote:
> 
> > most screen reader users will never be able to trigger the announcement of the accessible description, because there is no interactive element for them to navigate directly to
> 
> This is not quite right, because they would be able to focus it with Tab.  But it would still be random to encounter text in the focus order, and there would be no accessible role in the speech sequence to separate the description from the text of the element.
>
